### PR TITLE
Unit tests for evals functions

### LINKED ISSUE #298: Add unit tests for for evals functions
Add unit tests for calculate_evaluations_over_boolean_hypercube_for_eq, eval_eq, calculate_eq, and the transposed matrix multiplication helpers.

### PR DESCRIPTION
**Pull request by x-senpai-x**
resolves #298, #301

### CODE REVIEW

File: provekit/common/src/utils/sumcheck.rs [hunk 1]
```diff
@@ -113,6 +110,9 @@ pub fn calculate_evaluations_over_boolean_hypercube_for_eq(
     r: &[FieldElement],
     num_entries: usize,
 ) -> Vec<FieldElement> {
+    if num_entries == 0 {
```
Comment (x-senpai-x) [inline provekit/common/src/utils/sumcheck.rs:113]: if num_entries equals 0 the programme panics when it passes through `eval_eq`, so added this.

File: provekit/common/src/utils/sumcheck.rs [hunk 2]
```diff
@@ -113,6 +110,9 @@ pub fn calculate_evaluations_over_boolean_hypercube_for_eq(
...
+        return vec![];
+    }
     let full_size = 1usize << r.len();
     debug_assert!(num_entries <= full_size);
```
[the remark this answers is not in the document]
Comment (Bisht13) [reply provekit/common/src/utils/sumcheck.rs]: Agree with Copilot here.

File: provekit/common/src/utils/sumcheck.rs [hunk 3]
```diff
@@ -225,3 +225,194 @@ pub fn multiply_transposed_by_eq_alpha(
...
+
+    #[test]
+    fn test_eval_eq_truncated_left_only() {
+        // eval = [2,3,5], out has 1 slot (right_len = 0 each time), subtree_size = 8.
+        // Expected: eq([2,3,5], [0,0,0]) = (1-2)(1-3)(1-5) = (-1)(-2)(-4) = -8
+        let mut out = [FieldElement::zero()];
+        eval_eq(&[fe(2), fe(3), fe(5)], &mut out, FieldElement::one(), 8);
+        assert_eq!(out[0], fe(-8));
+    }
+
```
Comment (Bisht13) [inline provekit/common/src/utils/sumcheck.rs:341]: Both eval_eq tests exercise only the base case and the right_len == 0 branch. The main path where both halves of out have entries (line 147–150) is only covered indirectly via test_eq_hypercube_len4. Consider a direct test:

```rs
#[test]
fn test_eval_eq_both_halves() {
    let mut out = vec![FieldElement::zero(); 4];
    eval_eq(&[fe(2), fe(3)], &mut out, FieldElement::one(), 4);
    let expected = calculate_evaluations_over_boolean_hypercube_for_eq(&[fe(2), fe(3)], 4);
    assert_eq!(out, expected);
}
```

File: provekit/common/src/utils/sumcheck.rs [hunk 4]
```diff
@@ -225,3 +225,194 @@ pub fn multiply_transposed_by_eq_alpha(
...
+    #[test]
+    fn test_eq_hypercube_zero_entries() {
+        let result = calculate_evaluations_over_boolean_hypercube_for_eq(&[fe(2), fe(3), fe(5)], 0);
+        assert!(result.is_empty(), "non-empty r, zero entries");
+
+        let result = calculate_evaluations_over_boolean_hypercube_for_eq(&[], 0);
+        assert!(result.is_empty(), "empty r, zero entries");
+    }
+
```
Comment (Bisht13) [inline provekit/common/src/utils/sumcheck.rs:322]: Consider adding a num_entries == 1 test with non-empty r, this is the tightest truncation where only the first entry is computed and all recursion takes the left-only path. It's a distinct edge case from num_entries == 0:

```rs
#[test]
fn test_eq_hypercube_single_entry() {
    let r = [fe(2), fe(3)];
    let result = calculate_evaluations_over_boolean_hypercube_for_eq(&r, 1);
    assert_eq!(result, vec![calculate_eq(&r, &[fe(0), fe(0)])]);
}
```

**Review by Bisht13 — APPROVED** [review]
